astropy/time/core.py

```diff
@@ -1037,18 +1051,48 @@ def sort(self, axis=-1):
         return self[self._advanced_index(self.argsort(axis), axis,
                                          keepdims=True)]
 
+    @property
```

Comment:
This definition seemed a bit like a `lazyproperty`; it turns out that with that, the deleter is taken care of as well!
```
@lazyproperty
def cache(self):
    return defaultdict(dict)
```
and then:
```
In [1]: from astropy.time import Time

In [2]: t = Time.now()

In [3]: t.value
Out[3]: datetime.datetime(2016, 1, 4, 17, 47, 28, 515513)

In [4]: t.cache
Out[4]: defaultdict(<class 'dict'>, {'format': {'datetime': datetime.datetime(2016, 1, 4, 17, 47, 28, 515513)}})

In [5]: 'cache' in t.__dict__
Out[5]: True

In [6]: del t.cache

In [7]: 'cache' in t.__dict__
Out[7]: False
```